packages/shared/src/utils/currency/currency.ts

```diff
@@ -20,7 +42,7 @@ const crypto_currencies_display_order = [
     'USDK',
 ];
 
-export const reorderCurrencies = (list, type = 'fiat') => {
+export const reorderCurrencies = (list: Array<{ value: string; type: string }>, type: 'fiat' | string = 'fiat') => {
```

Comment:
@niloo-fs Not sure what should be the type here, But if you meant to set the type to be `fiat` or any other string then you should do as following:
```ts
'fiat' | Omit<string, 'fiat'>
```
Let me know if you need me to explain the reason for it 🙇🏻

Reply:
Could u pls explain it more?

Reply:
@niloo-fs Sure thing 🙇🏻 

So `string` type means any possible combination of characters, Right? So `string` already includes `fiat` as well.
When we say `'fiat | string`, TS tries to merge these two types together, And since `string` already includes `fiat` it will remove it and set the type to just be `string`.

So there is no difference between `fiat | string` and `string`, For that reason, we lose the autocomplete. To fix this we need to `Omit` the `fiat` from the string so now `string` can be any combination of characters except `fiat`.

What happens now is TS understands the value can either be `fiat` or any other `string` value and it can suggest `fiat` in the autocomplete.

If it still doesn’t make sense let me know to have a quick call 😅🙇🏻

Reply:
cool I got it @farzin-fs , thank u for your explanations, I will change them